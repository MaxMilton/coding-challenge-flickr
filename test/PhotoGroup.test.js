'use strict';

// set up mocks
jest.mock('../src/jsonp.js');

const PhotoGroup = require('../src/components/PhotoGroup.html');

describe('PhotoGroup component', () => {
  it('renders correctly', () => {
    expect.assertions(2);
    const target = document.createElement('div');
    function wrapper() {
      new PhotoGroup({ target });
    }
    expect(target.innerHTML).toMatchSnapshot();
    expect(wrapper).not.toThrow();
  });

  it('performs a search on suggestion button click', () => {
    expect.assertions(1);
    const target = document.createElement('div');
    new PhotoGroup({ target });
    const button = target.querySelector('.suggestions > .button');
    button.click();
    expect(target.innerHTML).toMatchSnapshot();
  });
});
