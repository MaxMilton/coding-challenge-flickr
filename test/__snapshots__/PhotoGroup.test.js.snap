// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGroup component performs a search on suggestion button click 1`] = `
"<div class=\\"photo-group\\"><a href=\\"https://www.flickr.com/photos/117671697@N03/42523418095/\\" class=\\"photo-item\\" rel=\\"noopener\\" target=\\"_blank\\"><div class=\\"photo-wrapper\\"><img src=\\"https://farm2.staticflickr.com/1763/42523418095_1dce215e24_m.jpg\\" class=\\"photo\\" alt=\\"Japan- Tokyo- Akihabara\\"></div>
  <div class=\\"photo-info\\"><div class=\\"photo-author\\">Author: venturidonatella</div>
    <div class=\\"photo-tags\\">Tags: tokyo japan giappone asia akihabara street strade streetlife streetscene persone people gentes portrait ritratto auto car automobile grattaceli colori colors nikon nikond500 d500 fumetti manga ritratti portraits</div></div>
  <div class=\\"photo-link\\"><i class=\\"icon-link\\"></i> See on Flickr</div></a><a href=\\"https://www.flickr.com/photos/pelican/43415352081/\\" class=\\"photo-item\\" rel=\\"noopener\\" target=\\"_blank\\"><div class=\\"photo-wrapper\\"><img src=\\"https://farm1.staticflickr.com/846/43415352081_266a75130f_m.jpg\\" class=\\"photo\\" alt=\\"From Hijiri-bashi bridge\\"></div>
  <div class=\\"photo-info\\"><div class=\\"photo-author\\">Author: pelican</div>
    <div class=\\"photo-tags\\">Tags: dscrx100 tokyo hijiribashi akihabara</div></div>
  <div class=\\"photo-link\\"><i class=\\"icon-link\\"></i> See on Flickr</div></a><a href=\\"https://www.flickr.com/photos/amcuk/28490047117/\\" class=\\"photo-item\\" rel=\\"noopener\\" target=\\"_blank\\"><div class=\\"photo-wrapper\\"><img src=\\"https://farm1.staticflickr.com/914/28490047117_4ab61e368d_m.jpg\\" class=\\"photo\\" alt=\\"Sonic the Hedgehog\\"></div>
  <div class=\\"photo-info\\"><div class=\\"photo-author\\">Author: AMcUK</div>
    <div class=\\"photo-tags\\">Tags: chiyodaku tōkyōto japan jp em10 omdem10 omdem10mkii em10mkii omd olympus olympusuk m43 micro43rds micro43 microfourthirds nippon tokyo akihabara sonic hedgehog sega sonicthehedgehog</div></div>
  <div class=\\"photo-link\\"><i class=\\"icon-link\\"></i> See on Flickr</div></a><a href=\\"https://www.flickr.com/photos/amcuk/41550051470/\\" class=\\"photo-item\\" rel=\\"noopener\\" target=\\"_blank\\"><div class=\\"photo-wrapper\\"><img src=\\"https://farm2.staticflickr.com/1765/41550051470_39862d1bbf_m.jpg\\" class=\\"photo\\" alt=\\"Cheese Hedgehog Cafe\\"></div>
  <div class=\\"photo-info\\"><div class=\\"photo-author\\">Author: AMcUK</div>
    <div class=\\"photo-tags\\">Tags: chiyodaku tōkyōto japan jp em10 omdem10 omdem10mkii em10mkii omd olympus olympusuk m43 micro43rds micro43 microfourthirds nippon tokyo akihabara hedgehog</div></div>
  <div class=\\"photo-link\\"><i class=\\"icon-link\\"></i> See on Flickr</div></a><a href=\\"https://www.flickr.com/photos/amcuk/28490042907/\\" class=\\"photo-item\\" rel=\\"noopener\\" target=\\"_blank\\"><div class=\\"photo-wrapper\\"><img src=\\"https://farm1.staticflickr.com/927/28490042907_f8b87d4465_m.jpg\\" class=\\"photo\\" alt=\\"Akihabara sunset\\"></div>
  <div class=\\"photo-info\\"><div class=\\"photo-author\\">Author: AMcUK</div>
    <div class=\\"photo-tags\\">Tags: chiyodaku tōkyōto japan jp em10 omdem10 omdem10mkii em10mkii omd olympus olympusuk m43 micro43rds micro43 microfourthirds nippon tokyo akihabara sunset urban streetscene street</div></div>
  <div class=\\"photo-link\\"><i class=\\"icon-link\\"></i> See on Flickr</div></a></div><!---->"
`;

exports[`PhotoGroup component renders correctly 1`] = `""`;
